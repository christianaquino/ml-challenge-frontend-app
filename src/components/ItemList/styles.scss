.itemlist {
  background-color: white;
  grid-area: main;
  padding: 0 12px;
}
