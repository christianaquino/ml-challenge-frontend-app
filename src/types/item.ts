export interface IPrice {
  currency: string,
  amount: Number
}

export interface IItem {
  id: string,
  picture: string,
  title: string,
  price: IPrice
  condition: string
  free_shipping: boolean
}

export interface IItemDetails extends IItem {
  sold_quantity: Number,
  description: string
}
