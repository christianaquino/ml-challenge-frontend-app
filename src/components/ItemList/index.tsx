import Item from '../Item';
import { IItem } from '../../types/item';

import './styles.scss';

interface IItemListProps {
    items: IItem[]
}

function ItemList({ items }: IItemListProps) {
  return (
    <div className="itemlist">
      {items.map((item: IItem) => <Item item={item} key={item.id} />)}
    </div>
  );
}

export default ItemList;
