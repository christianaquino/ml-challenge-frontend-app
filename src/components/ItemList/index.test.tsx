import { render } from '@testing-library/react';
import { ClassAttributes, AnchorHTMLAttributes } from 'react';
import ItemList from '.';
import { IItem } from '../../types/item';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  Link: (
    // eslint-disable-next-line no-undef
    props: JSX.IntrinsicAttributes
      & ClassAttributes<HTMLAnchorElement>
      & AnchorHTMLAttributes<HTMLAnchorElement>,
  ) => (
    // eslint-disable-next-line react/jsx-props-no-spreading
    <a {...props}>
      { // eslint-disable-next-line react/destructuring-assignment
          props.children
        }
    </a>
  ),
}));
describe('ItemList component', () => {
  test('renders correctly', () => {
    const item1: IItem = {
      id: 'MLC8752432',
      picture: '//item.jpg',
      title: 'IPhone 12',
      price: {
        amount: 650000,
        currency: 'CLP',
      },
      condition: 'Nuevo',
      free_shipping: true,
    };

    const item2 = { ...item1, id: 'MLC8752433' };
    const item3 = { ...item1, id: 'MLC8752434', free_shipping: false };
    const { asFragment } = render(<ItemList items={[item1, item2, item3]} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
