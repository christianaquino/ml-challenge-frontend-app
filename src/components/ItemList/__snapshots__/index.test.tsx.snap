// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="itemlist"
  >
    <a
      class="item"
      to="/items/MLC8752432"
    >
      <div
        class="item--imagecontainer"
      >
        <img
          alt="IPhone 12"
          class="item--image"
          src="//item.jpg"
        />
      </div>
      <div
        class="item--price"
      >
        CLP 650000
      </div>
      <div
        class="item--title"
      >
        IPhone 12
      </div>
      <div
        class="item--condition"
      >
        Nuevo
      </div>
      <div
        class="item--shipping-free"
      >
        Envío Gratis
      </div>
    </a>
    <a
      class="item"
      to="/items/MLC8752433"
    >
      <div
        class="item--imagecontainer"
      >
        <img
          alt="IPhone 12"
          class="item--image"
          src="//item.jpg"
        />
      </div>
      <div
        class="item--price"
      >
        CLP 650000
      </div>
      <div
        class="item--title"
      >
        IPhone 12
      </div>
      <div
        class="item--condition"
      >
        Nuevo
      </div>
      <div
        class="item--shipping-free"
      >
        Envío Gratis
      </div>
    </a>
    <a
      class="item"
      to="/items/MLC8752434"
    >
      <div
        class="item--imagecontainer"
      >
        <img
          alt="IPhone 12"
          class="item--image"
          src="//item.jpg"
        />
      </div>
      <div
        class="item--price"
      >
        CLP 650000
      </div>
      <div
        class="item--title"
      >
        IPhone 12
      </div>
      <div
        class="item--condition"
      >
        Nuevo
      </div>
    </a>
  </div>
</DocumentFragment>
`;
